### ISSUE TITLE
Installation issue with cpp functions

### ISSUE THREAD

**Issue by Chris-Cherry** [amcdavid/HurdleNormal#6]
I'm attempting to install HurdleNormal and am having issues with cpp functions missing. I'm working on Ubuntu 18.04.5 with R 3.6.3. I've tried installation through both github and bioconductor and both run into the same issue. Code and error below:

```{R}
R version 3.6.3 (2020-02-29) -- "Holding the Windsock"
Copyright (C) 2020 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> devtools::install_github('amcdavid/HurdleNormal')
Downloading GitHub repo amcdavid/HurdleNormal@master
These packages have more recent versions available.
Which would you like to update?

 1: All                                             
 2: CRAN packages only                              
 3: None                                            
 4: Rcpp         (1.0.3       -> 1.0.5      ) [CRAN]
 5: RcppArmad... (0.9.850.1.0 -> 0.9.900.3.0) [CRAN]
 6: nloptr       (1.2.2.1     -> 1.2.2.2    ) [CRAN]
 7: reshape2     (1.4.3       -> 1.4.4      ) [CRAN]
 8: data.table   (1.12.8      -> 1.13.0     ) [CRAN]
 9: plyr         (1.8.5       -> 1.8.6      ) [CRAN]
10: foreach      (1.4.8       -> 1.5.0      ) [CRAN]
11: glue         (1.3.1       -> 1.4.2      ) [CRAN]
12: stringi      (1.4.5       -> 1.5.3      ) [CRAN]
13: digest       (0.6.23      -> 0.6.25     ) [CRAN]
14: isoband      (0.2.0       -> 0.2.2      ) [CRAN]
15: rlang        (0.4.6       -> 0.4.7      ) [CRAN]
16: scales       (1.1.0       -> 1.1.1      ) [CRAN]
17: tibble       (3.0.1       -> 3.0.3      ) [CRAN]
18: withr        (2.1.2       -> 2.2.0      ) [CRAN]
19: testthat     (2.3.1       -> 2.3.2      ) [CRAN]
20: cli          (2.0.1       -> 2.0.2      ) [CRAN]
21: ellipsis     (0.3.0       -> 0.3.1      ) [CRAN]
22: pkgload      (1.0.2       -> 1.1.0      ) [CRAN]
23: pkgbuild     (1.0.6       -> 1.1.0      ) [CRAN]
24: rstudioapi   (0.10        -> 0.11       ) [CRAN]
25: callr        (3.4.1       -> 3.4.4      ) [CRAN]
26: backports    (1.1.5       -> 1.1.9      ) [CRAN]
27: processx     (3.4.1       -> 3.4.4      ) [CRAN]
28: ps           (1.3.0       -> 1.3.4      ) [CRAN]
29: pillar       (1.4.3       -> 1.4.6      ) [CRAN]
30: vctrs        (0.3.0       -> 0.3.4      ) [CRAN]

Enter one or more numbers, or an empty line to skip updates:

✔  checking for file ‘/tmp/Rtmp8G3VLf/remotes1209b161184db/amcdavid-HurdleNormal-f91e584/DESCRIPTION’
─  preparing ‘HurdleNormal’:
✔  checking DESCRIPTION meta-information ...
─  cleaning src
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
   Removed empty directory ‘HurdleNormal/inst/aoas_simulations’
   Removed empty directory ‘HurdleNormal/inst’
─  looking to see if a ‘data/datalist’ file should be added
─  building ‘HurdleNormal_0.98.7.tar.gz’
   
Installing package into ‘/home/chonky/R/x86_64-pc-linux-gnu-library/3.6’
(as ‘lib’ is unspecified)
* installing *source* package ‘HurdleNormal’ ...
** using staged installation
** libs
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I"/home/chonky/R/x86_64-pc-linux-gnu-library/3.6/RcppArmadillo/include" -I"/home/chonky/R/x86_64-pc-linux-gnu-library/3.6/Rcpp/include"   -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-V28x5H/r-base-3.6.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c hurdle_distributions.cpp -o hurdle_distributions.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I"/home/chonky/R/x86_64-pc-linux-gnu-library/3.6/RcppArmadillo/include" -I"/home/chonky/R/x86_64-pc-linux-gnu-library/3.6/Rcpp/include"   -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-V28x5H/r-base-3.6.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c hurdle_likelihood.cpp -o hurdle_likelihood.o
g++ -std=gnu++11 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o HurdleNormal.so hurdle_distributions.o hurdle_likelihood.o -L/usr/lib/R/lib -lR
installing to /home/chonky/R/x86_64-pc-linux-gnu-library/3.6/00LOCK-HurdleNormal/00new/HurdleNormal/libs
** R
** data
*** moving datasets to lazyload DB
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
** checking absolute paths in shared objects and dynamic libraries
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (HurdleNormal)
> library(HurdleNormal)
> library(GGally)
Loading required package: ggplot2
Registered S3 method overwritten by 'GGally':
  method from   
  +.gg   ggplot2
> library(ggplot2)
> G = matrix(c(-16, 2, 0,
+              2, -17, 2,
+              0, 2, -16), nrow=3)
> H = matrix(c(5, 0, 0,
+              0, 5, 0,
+              0, 0, 5), nrow=3)
> K = diag(1, nrow=3)
> y = as.data.frame(rGibbsHurdle(G, H, K, 301000, thin = .04, burnin=1000))
Error in .Call("_HurdleNormal_cpp_rGibbsHurdle", PACKAGE = "HurdleNormal",  : 
  "_HurdleNormal_cpp_rGibbsHurdle" not available for .Call() for package "HurdleNormal"
```

**Comment by amcdavid** [comment]
Thanks, I am able to reproduce this with R 3.6.  This seems to be due to a mismatch between the Rcpp version that I used when developing vs the version you are using (eg, everything works fine in R 4.0).  For a temporary work-around, if you clone the repository and then call devtools::document(), devtools::install() that should fix the issue.  

However, I will try to find a fix so that it installs cleanly from github regardless of Rcpp version.

**Comment by Chris-Cherry** [comment]
Hey David,

That workaround worked for me. Thanks!

Chris

**Comment by amcdavid** [comment]
Also closed in 11dc0b1
